RedHatInsights/ocp-advisor-frontend
# feat(OCPADVISOR-71): Show upgrade risks table

Implements https://issues.redhat.com/browse/OCPADVISOR-71.
Mocked at https://www.sketch.com/s/c5c07e06-f4a1-4c64-9cdf-8b12d8ac9ad5/a/25OY5ve.

This introduces the upgrade risks table that is shown under the "Upgrade risks" tab on the cluster details view. 

- The data for this table are fetched from the new back-end (Smart proxy) endpoint `v1/cluster/%id/upgrade-risks-prediction`.
- The table must be shown if there are firing alerts or cluster operators available.
- When there are no firing alerts nor cluster operators, the tab must show successful empty message.
- If the response from back end is 503, then the tab must show the "not available" empty state.
- For all other edge cases that are unknown/unexpected, the tab must show "something went wrong."  

## How to test

The real endpoint will serve no data for real clusters, thus it is better to run this PR together with the mock service: 
1. Follow https://github.com/RedHatInsights/ocp-advisor-frontend#using-insights-results-aggregator-mock instructions and make sure the webpack server and the mock server are running.
2. Use different cluster IDs from https://github.com/RedHatInsights/insights-results-aggregator-mock/#clusters-that-return-valid-data and navigate to `https://stage.foo.redhat.com:1337/beta/openshift/insights/advisor/clusters/%ID`.
3. Click on the "Upgrade risks" tab.
4. Make sure the content matches with the specifics of this cluster (e.g., `00000003-eeee-eeee-eeee-000000000001` must show firing alerts and cluster operators).

## Screenshots

![image](https://user-images.githubusercontent.com/31385370/231707569-8ba19375-e5b6-4e05-9bd5-539702d9cf2d.png)

![image](https://user-images.githubusercontent.com/31385370/231708554-a399762d-8672-420e-a656-3145263f2ae6.png)

![image](https://user-images.githubusercontent.com/31385370/231768287-7c74c2e0-2897-4987-ac52-2c1beab19716.png)

![image](https://user-images.githubusercontent.com/31385370/231768651-68f3a166-ae5b-4680-b856-a8403149511b.png)

![image](https://user-images.githubusercontent.com/31385370/231769347-2f1cd138-12ee-4432-a9e6-93f0732dfbf6.png)
